### PR TITLE
admins can filter participants by type

### PR DESCRIPTION
**Pull request by asmega**
## Ticket and context

Ticket: A tiny part of https://trello.com/c/3FttpebV/406-handle-duplicate-signups

### Screenshots

Desktop
Screenshot 2021-09-09 at 12 14 00

Mobile
Screenshot 2021-09-09 at 12 14 46

### Code quality checks
- [x] All commit messages are meaningful and true
- [x] Added enough automated tests

### HTML Checks
- [x] All new pages have automated accessibility checks
- [x] All new pages have visual tests via Percy

### Gotchas
- [x] All `School` queries are correctly scoped - `eligible` when they need to be

### How can someone see it it review app?

- login as an admin
- filter participants by different types
- should only return participants of selected type

## External API changes

- none

### CODE REVIEW

File: app/components/search_box.html.erb [hunk 1]
```diff
@@ -9,6 +9,20 @@
...
+        [
+          OpenStruct.new(id: "", name: ""),
+          OpenStruct.new(id: "ParticipantProfile::ECT", name: "ECT"),
+          OpenStruct.new(id: "ParticipantProfile::Mentor", name: "Mentor"),
+          OpenStruct.new(id: "ParticipantProfile::NPQ", name: "NPQ")
+        ],
+        :id,
+        :name,
+        label: { text: "Filter by type" } %>
+    </div>
```
Comment (kicferk1) [inline app/components/search_box.html.erb:24]: Can we make it stay after search, like query?

Comment (asmega) [reply app/components/search_box.html.erb:24]: Yes. Actually this has pointed me to another bug whereby this component is shared on other views such as schools. so let me fix that too

Comment (asmega) [reply app/components/search_box.html.erb]: Ok, filters are now passed in which solves the "used in different area problems". One can now specify the value so it'll get passed down to the filter to select it.

**Review by kicferk1 — APPROVED** [review]

**Review by jamesrose — APPROVED** [review]
Yes!

**Pull request merged**